.github/workflows/snapshot_release.yml

```diff
@@ -0,0 +1,60 @@
+name: snapshot_release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        required: true
+        type: string
+
+jobs:
+  install:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/install_with_cache
+  build:
+    runs-on: ubuntu-latest
+    needs:
+      - install
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/build_with_cache
+  test:
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/restore_build_cache
+      - run: npm run set-script-shell
+      - run: npm run test
+  check_package_versions:
+    runs-on: ubuntu-latest
+    needs:
+      - install
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/restore_install_cache
+      - run: npm run check:package-versions
+  publish_snapshot:
+    if: ${{ github.ref_name != 'main' }}
```

Comment:
is the intended behavior here that snapshot releases must be from a separate branch?

Reply:
Correct. https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md#what-to-do-with-the-snapshot-branch

Reply:
On the other hand. After reading this once more.
I don't see a reason to exclude main. 
The version changes are never merged back. I'll remove this.